src/web/page/web_page.rs

```diff
@@ -214,5 +245,19 @@ pub(crate) async fn render_templates_middleware<B>(
         .nonce()
         .to_owned();
 
-    render_response(next.run(req).await, &templates.templates, &csp_nonce)
+    let response = next.run(req).await;
+
+    let cpu_intensive_rendering: bool = response
+        .extensions()
+        .get::<DelayedTemplateRender>()
+        .map(|render| render.cpu_intensive_rendering)
+        .unwrap_or(false);
```

Comment:
I didn't find a nice way to combine the optional `spawn_blocking` without repeating `response.extensions().get()` because of the recursion inside `render_response`. `async_recursion` didn't help, though I might miss some async specifics. 

I'm happy to take hints :) 

Or we leave it like this.

Reply:
I played around with it a bit and managed to get a working version: <https://github.com/Nemo157/docs.rs/commit/ff545973d33aa60bc371b0852830d6b1a1645fd9>. Main thing to allow recursive async-fn is to not use the `async fn` sugar and have the signature just as `fn() -> BoxFuture`.

Reply:
Thanks, I didn't know that part. 

I cherry-picked your commit and added it to this branch.

Reply:
Also this got rid of the `.expect` :)